WooCommerce/Classes/ViewRelated/Dashboard/Blaze/BlazeCampaignDashboardView.swift

```diff
@@ -25,8 +25,8 @@ final class BlazeCampaignDashboardViewHostingController: SelfSizingHostingContro
         fatalError("init(coder:) has not been implemented")
     }
 
-    override func viewWillAppear(_ animated: Bool) {
-        super.viewWillAppear(animated)
+    override func viewDidLoad() {
+        super.viewDidLoad()
         reload()
     }
```

Comment:
💡 Instead of reloading the Blaze campaign here, we should observe `ServiceLocator.stores.site` and trigger reloading when it changes. Currently, when I switch sites, the section disappears:

https://github.com/woocommerce/woocommerce-ios/assets/5533851/4bf77115-a279-44ea-8ce6-bdcb16905bbf

Reply:
The section doesn't show up when I first log in either.

Reply:
Good catch and thanks for the idea. 🙇 

I followed the blaze banner's refresh pattern in https://github.com/woocommerce/woocommerce-ios/pull/10962/commits/3f4e89994e2b7615b70de73cd04aaf2325282874